clang/lib/Driver/Driver.cpp

```diff
@@ -3787,6 +3785,53 @@ class OffloadingActionBuilder final {
       SYCLDeviceActions.clear();
     }
 
+    void addSYCLDeviceLibs(const ToolChain *TC, ActionList &DeviceLinkObjects,
+                           bool isSpirvAOT, bool isMSVCEnv) {
+      enum SYCLDeviceLibType {
+        sycl_devicelib_wrapper,
+        sycl_devicelib_fallback
+      };
+      StringRef LibLoc, LibSysUtils;
+      if (isMSVCEnv) {
+        LibLoc = Args.MakeArgString(TC->getDriver().Dir + "/../bin");
+        LibSysUtils = "libsycl-msvc";
```

Comment:
Why are the device libs for Windows built in the `bin` directory as opposed to the `lib` directory like Linux?

Reply:
I believe archive libraries on Windows are not supposed to be in `bin` directory.  `bin` is for DLLs and executables.  For OpenMP the `BC` libraries are located in `lib` (as they are considered archive librareis), and `SPV` libraries are located in `bin` (as the are considered 'dynamic' libraries, i.e. DLLs).  Anyway, I think the `BC` libraries must not be located in `bin`.

Reply:
Hi, @mdtoguchi and @vzakhari 
Currently, the device libraries on Windows platform are located in bin/ folder for DPC++ compiler. You can find the details in :
https://github.com/intel/llvm/blob/sycl/libdevice/cmake/modules/SYCLLibdevice.cmake#L209
So, I followed it in this patch. I think the reason why put device libraries in bin/ on Windows is we want to put all 'sycl-related' libraries together with libsycl.so/dll? @asavonic , please feel free to correct me.
Thank you very much.